ambry-clustermap/src/main/java/com.github.ambry.clustermap/CompositeClusterManager.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.codahale.metrics.MetricRegistry;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * A cluster manager that is a wrapper over a {@link StaticClusterManager} instance and a {@link HelixClusterManager}
+ * instance and uses the {@link StaticClusterManager} as the source-of-truth. It relays events to both and checks for
+ * and reports inconsistencies in the views from the two underlying cluster managers.
+ */
+class CompositeClusterManager implements ClusterMap {
+  private final StaticClusterManager staticClusterManager;
+  private final HelixClusterManager helixClusterManager;
+  private final HelixClusterManagerMetrics helixClusterManagerMetrics;
+
+  /**
+   * Construct a CompositeClusterManager instance.
+   * @param staticClusterManager the {@link StaticClusterManager} instance to use as the source-of-truth.
+   * @param helixClusterManager the {@link HelixClusterManager} instance to use for comparison of views.
+   */
+  CompositeClusterManager(StaticClusterManager staticClusterManager, HelixClusterManager helixClusterManager) {
+    this.staticClusterManager = staticClusterManager;
+    this.helixClusterManager = helixClusterManager;
+    this.helixClusterManagerMetrics = helixClusterManager.helixClusterManagerMetrics;
+  }
+
+  @Override
+  public PartitionId getPartitionIdFromStream(DataInputStream stream) throws IOException {
+    return staticClusterManager.getPartitionIdFromStream(stream);
```

Comment:
no comparison required here? Can this fail only due to programming errors and not due to innate differences b/w the two impls?

For e.g., could this fail because a partition exists in the static manager but not in the helix manager? Is that checked on bootstrap (or during load)?